src/unaudited/ECDSAStakeRegistry.sol

```diff
@@ -0,0 +1,385 @@
+    /// @param _quorum The new quorum configuration
+    /// @return bool Indicates if the quorum is valid
+    function _isValidQuorum(Quorum memory _quorum) internal pure returns (bool) {
+        StrategyParams[] memory strategies = _quorum.strategies;
+        address lastStrategy;
+        address currentStrategy;
+        uint256 totalMultiplier;
+        for (uint256 i; i < strategies.length; i++) {
+            currentStrategy = address(strategies[i].strategy);
+            if (lastStrategy >= currentStrategy) revert NotSorted();
+            lastStrategy = currentStrategy;
+            totalMultiplier += strategies[i].multiplier;
+        }
+        if (totalMultiplier != BPS) return false;
+        return true;
+    }
+
+    /**
+     * @notice Common logic to verify a batch of ECDSA signatures against a hash, using either last stake weight or at a specific block.
+     * @param _dataHash The hash of the data the signers endorsed.
+     * @param _signers A collection of addresses that endorsed the data hash.
+     * @param _signatures A collection of signatures matching the signers.
+     * @param _referenceBlock The block number for evaluating stake weight; use max uint32 for latest weight.
+     */
+    function _checkSignatures(
+        bytes32 _dataHash,
+        address[] memory _signers,
+        bytes[] memory _signatures,
+        uint32 _referenceBlock
+    ) internal view {
+        uint256 signersLength = _signers.length;
+        address lastSigner;
+        uint256 signedWeight;
+
+        _validateSignaturesLength(signersLength, _signatures.length);
+        for (uint256 i; i < signersLength; i++) {
+            address currentSigner = _signers[i];
+
+            _validateSortedSigners(lastSigner, currentSigner);
+            _validateSignature(currentSigner, _dataHash, _signatures[i]);
+
+            lastSigner = currentSigner;
+            uint256 operatorWeight = _getOperatorWeight(currentSigner, _referenceBlock);
+            signedWeight += operatorWeight;
+        }
+
+        _validateThresholdStake(signedWeight, _referenceBlock);
+    }
+
+    /// @notice Validates that the number of signers equals the number of signatures, and neither is zero.
+    /// @param _signersLength The number of signers.
+    /// @param _signaturesLength The number of signatures.
+    function _validateSignaturesLength(
+        uint256 _signersLength,
+        uint256 _signaturesLength
+    ) internal pure {
+        if (_signersLength != _signaturesLength) revert LengthMismatch();
+        if (_signersLength == 0) revert InvalidLength();
+    }
+
+    /// @notice Ensures that signers are sorted in ascending order by address.
+    /// @param _lastSigner The address of the last signer.
+    /// @param _currentSigner The address of the current signer.
+    function _validateSortedSigners(address _lastSigner, address _currentSigner) internal pure {
+        if (_lastSigner >= _currentSigner) revert NotSorted();
```

Comment:
you want strict inequality here to prevent duplicates, not non-strict inequality

Reply:
ah, mb, it's just from custom error logic being flipped